core/src/main/java/com/amplifyframework/api/graphql/Page.java

```diff
@@ -0,0 +1,23 @@
+package com.amplifyframework.api.graphql;
+
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.api.ApiException;
+import com.amplifyframework.core.Consumer;
+
+public abstract class Page<T> {
+    private final GraphQLResponse<Iterable<T>> response;
+
+    public abstract boolean hasNextPage();
+
+    public abstract GraphQLOperation<T> queryNextPage(@NonNull Consumer<Page<T>> onResponse,
+                                                      @NonNull Consumer<ApiException> onFailure);
+
+    protected Page(GraphQLResponse<Iterable<T>> response) {
+        this.response = response;
+    }
+
+    public GraphQLResponse<Iterable<T>> getResponse() {
```

Comment:
`final` to close the getter. Can this be just `protected`?

Reply:
> `final` to close the getter. 

Addressed in https://github.com/aws-amplify/amplify-android/pull/488

> Can this be just `protected`?

No, this is designed to be exposed - this is the only way a customer would be able to access the `GraphQLResponse`, in order to access `data` and `errors`.